src/main/java/com/worldwarofants/game/arch/console/ConsoleOutputHandler.java

```diff
@@ -24,17 +31,28 @@ public static void spell(String input) {
                 Thread.currentThread().interrupt();
```

Comment:
Do you think that interrupting here is correct? What happened with the SPELL_DELAY? I can't see it being used anywhere.

Reply:
I'm a bit confused. I didn't write 'Thread.currentThread().interrupt();', it was already there. 

Is the SPELL-DELAY not used when the 'spell' method is called in the New Game View?

Reply:
That is the idea. The delay needs to be used in the spell method, to delay every character that is being displayed, to simulate a typing effect. Interrupt() there is a mistake, you need to replace it with something that will achieve that typing effect.

Read a little bit about what threads are in programming, and how to do some basic operations with them. Hint: delaying execution of code.

Reply:
Ok thanks, I'll look into it. I'm just letting you know that I did not put Interrupt() there. I did not write any of that code, so i didn't want to go about deleting it.

Reply:
Yeah I can see that you didn't do it. I'm not sure why it's there though, it doesn't make much sense. So we should probably fix it.

Reply:
Hey this hasn't changed since we last reviewed it? It's there to handle the Interrupted exception if thrown by the .sleep above.

Reply:
Oh my god thanks for making me look into it. Github had folded the code for me and I totally missed the sleep. 

@AdamPhilipSmith You don't have to do anything about this, it's working, I'm just blind.

@liam027 How does does interrupting the thread handle the exception? Isn't the exception happening because the thread was already interrupted?

Reply:
From my limited understanding of exception handling and threads, it goes like this: if Interrupted while sleeping the thread gets an Interrupted flag and throws the exception. Calling interrupt() on it actually clears (in this case I think toggles) the flag and it is nolonger set to "interrupted" and will just continue what it was doing.

Reply:
Your understanding is correct. But isn't that the same as just ignoring the exception? I'm not sure that anything changes, with or without that line of code.

I don't think that flag is ever checked internally, if I recall correctly. The interrupt flag is there for people to use it to handle such interruptions. 

For example, if you're saving the game in a save file, and the thread gets interrupted, you should probably delete the half completed file before closing the thread, if the file is gonna be useless without it being fully finished. 

Basically, the interruption mechanism is there to allow you to gracefully finish your threads halfway through, without having your code break in weird ways.